sonar-go-to-slang/build.gradle.kts

```diff
@@ -14,85 +14,97 @@
+            }
         )
+
+        outputs.file("goparser_generated.go")
+        outputs.dir("build/executable")
+        outputs.cacheIf { true }
+
+        callMake("build")
     }
-}
 
-val compileGo by tasks.registering(Exec::class) {
-    group = "build"
-    description = "Generate Go parser and build the Go executable"
-    inputs.files(
-        fileTree(projectDir).matching {
-            include(
-                "**/*.go",
-                "**/go.mod",
-                "**/go.sum",
-                "make.bat",
-                "make.sh"
-            )
-            exclude("build/**", "*_generated.go")
-        }
-    )
-    outputs.file("goparser_generated.go")
-    outputs.dir("build/executable")
-    outputs.cacheIf { true }
+    tasks.register<Exec>("lintGoCode") {
+        description = "Run an external Go linter."
+        group = "verification"
 
-    callMake("build")
-}
+        val reportPath = layout.buildDirectory.file("reports/golangci-lint-report.xml")
+        inputs.files(
+            fileTree(projectDir).matching {
+                include("src/**/*.go")
+            }
+        )
+        outputs.files(reportPath)
+        outputs.cacheIf { true }
 
-val testGo by tasks.registering(Exec::class) {
-    group = "verification"
-    description = "Generate Go test report"
-    inputs.dir("build/executable")
+        commandLine(
+            "golangci-lint",
+            "run",
+            "--go=${requireNotNull(System.getenv("GO_VERSION")) { "Go version is unset in the environment" }}",
+            "--out-format=checkstyle:${reportPath.get().asFile}"
+        )
+        // golangci-lint returns non-zero exit code if there are issues, we don't want to fail the build in this case.
+        // A report with issues will be later ingested by SonarQube.
+        isIgnoreExitValue = true
+    }
 
-    callMake("generate-test-report")
-}
+    tasks.register<Exec>("testGoCode") {
+        description = "Test the executable produced by the compile go code step."
+        group = "verification"
 
-val cleanGo by tasks.registering(Exec::class) {
-    group = "build"
-    description = "Clean the Go build"
+        dependsOn("compileGo")
+        callMake("test")
+    }
 
-    callMake("clean")
-}
+    tasks.named("clean") {
+        dependsOn("cleanGoCode")
+    }
 
-tasks.clean {
-    dependsOn(cleanGo)
-}
-testGo {
-    dependsOn(compileGo)
-}
-tasks.build {
-    dependsOn(testGo)
-}
+    tasks.named("assemble") {
+        dependsOn("compileGo")
+    }
 
-spotless {
-    java {
-        // No Java sources in this project
-        target("")
+    tasks.named("test") {
+        dependsOn("testGoCode")
     }
-}
 
-if (System.getenv("CI") == "true") {
-    // spotless is enabled only for CI, because spotless relies on Go installation being available on the machine
     spotless {
         go {
             val goVersion = providers.environmentVariable("GO_VERSION").getOrElse("1.23.4")
```

Comment:
Here is another place where the version is mentioned. Maybe we can extract it as a constant and use GO_VERSION or this constant, and then pass it as build arg to Docker